routes/project/github.rb

```diff
@@ -12,78 +12,83 @@ class Clover
       authorize("Project:github", @project.id)
 
       r.get true do
-        if @project.github_installations.empty?
-          r.redirect "#{@project.path}/github/setting"
+        if (installation = @project.github_installations.first)
+          r.redirect "#{@project.path}/github/#{installation.ubid}/runner"
         end
-        r.redirect "#{@project.path}/github/runner"
+        view "github/index"
       end
 
-      r.on "runner" do
-        r.get true do
-          @runners = Serializers::GithubRunner.serialize(@project.github_runners_dataset.eager(:vm, :strand).reverse(:created_at).all)
-
-          view "github/runner"
+      r.get "create" do
+        unless @project.has_valid_payment_method?
+          flash["error"] = "Project doesn't have valid billing information"
+          r.redirect "#{@project.path}/github"
         end
+        session[:github_installation_project_id] = @project.id
 
-        r.is String do |runner_ubid|
-          next unless (runner = GithubRunner.from_ubid(runner_ubid)) && runner.installation.project_id == @project.id
-
-          r.delete true do
-            runner.incr_skip_deregistration
-            runner.incr_destroy
-            flash["notice"] = "Runner '#{runner.ubid}' forcibly terminated"
-            204
-          end
-        end
+        r.redirect "https://github.com/apps/#{Config.github_app_name}/installations/new", 302
       end
 
-      r.get "setting" do
-        @installations = Serializers::GithubInstallation.serialize(@project.github_installations)
+      r.on String do |installation_ubid|
+        next unless (@installation = GithubInstallation.from_ubid(installation_ubid)) && @installation.project_id == @project.id
 
-        view "github/setting"
-      end
+        r.get "setting" do
+          view "github/setting"
+        end
 
-      r.on "installation" do
-        r.get "create" do
-          unless @project.has_valid_payment_method?
-            flash["error"] = "Project doesn't have valid billing information"
-            r.redirect "#{@project.path}/github"
+        r.post true do
+          unless r.params["cache_enabled"].nil?
+            @installation.cache_enabled = r.params["cache_enabled"] == "true"
+            flash["notice"] = "Transparent cache is #{@installation.cache_enabled ? "enabled" : "disabled"}"
           end
-          session[:github_installation_project_id] = @project.id
 
-          r.redirect "https://github.com/apps/#{Config.github_app_name}/installations/new", 302
+          unless r.params["performance_runner_enabled"].nil?
+            @installation.allocator_preferences["family_filter"] = if r.params["performance_runner_enabled"] == "true"
+              ["performance", "standard"]
+            end
+            @installation.modified!(:allocator_preferences)
```

Comment:
Hey Enes, I'll try to address the entire UI's content in one comment. If you'd like, happy to also chat over a quick call.

* Ultra Performant Runners -> High Performance Runners
* Speed up your builds with cutting edge gaming CPUs! We automatically route your jobs to premium machines. If capacity's full, you seamlessly fall back to standard runners with no extra wait time. You always get billed by the minute and for the runner type you're using.
* What's different? Fast gaming CPUs, Twice as fast, One fifth the cost, 10x better price/performance
* Boost your builds. 1/5th the cost of GitHub runners

It's hard for me to reason about how the content would look visually. Make when you have a few alternatives, we could walk over them together.